File: src/main/java/net/littlelite/smartrest/exceptions/ResourceAlreadyExists.java
/*
 * Project SmartREST
 * Copyright (c) Alessio Saltarin 2021.
 * This software is licensed under MIT License (see LICENSE)
 */

package net.littlelite.smartrest.exceptions;

public class ResourceAlreadyExists extends RuntimeException
{
    public ResourceAlreadyExists(String id)
    {
        super("Resource with ID: '" + id + "' already exists.");
    }

    public ResourceAlreadyExists(int id)
    {
        super("Resource with ID: '" + id + "' already exists.");
    }
}

File: src/main/kotlin/net/littlelite/smartrest/exceptions/ResourceAlreadyExists.kt
/*
 * Project SmartREST
 * Copyright (c) Alessio Saltarin 2022-23
 * This software is licensed under MIT License (see LICENSE)
 */


package net.littlelite.smartrest.exceptions

class ResourceAlreadyExists : RuntimeException
{
    constructor(id: String) : super("Resource with ID: '$id' already exists.")
    constructor(id: Int) : super("Resource with ID: '$id' already exists.")
}

